tests/test-cmd/test.build.js

```diff
@@ -37,6 +37,40 @@ describe('build', () => {
     );
   });
 
+  it('zips a package with custom file extension', () => {
+    return withTempDir(
+      (tmpDir) =>
+        build({
+          sourceDir: fixturePath('minimal-web-ext'),
+          artifactsDir: tmpDir.path(),
+          extension: 'blah',
+        })
+        .then((buildResult) => {
+          assert.match(buildResult.extensionPath,
+                       /minimal_extension-1\.0\.blah$/);
+          return buildResult.extensionPath;
+        })
+    );
+  });
+
+  it('throws an error when invalid extension is provided', () => {
+    let failed = false;
+    try {
```

Comment:
after that change, here's [the pattern](https://github.com/rhelmer/web-ext/blob/3556526a4c2c1f357265982a8ab96cfb2b5ebfe9/tests/test-cmd/test.build.js#L203-L208) we use to check for promise rejections.